docker/deploy.sh

```diff
@@ -1,56 +1,127 @@
-
-    sed -i.bak s/{YOUR-DOMAIN-HERE}/$domain_name/g docker-compose.yml
-
-    docker stack deploy -c docker-compose.yml op-scim
-    mv docker-compose.yml.bak docker-compose.yml
-
-    docker service logs --raw -f op-scim_scim
-}
-
-echo 'Please ensure you have read the README before continuing.'
-echo 'Which docker deployment method are you using?'
-read -p '[compose] or [swarm]: ' docker_path
-
-if [ "$docker_path" == "compose" ];
-then 
-    run_docker_compose
-elif [ "$docker_path" == "swarm" ]
+#!/usr/bin/env bash
+
+# Docker Swarm deployment script
+# Please ensure you've read PREPARING.md and docker/README.md
+
+# set the full path of the docker examples directory
+docker_path=$(dirname $(realpath $0))
+
+echo "Initiating 1Password SCIM Bridge Deployment to Docker Swarm"
+echo " "
+echo "Please specify the following options."
+
+while ! [[ "$docker_type" =~ ^(swarm|compose)$ ]]; do
+    read -p "Docker Swarm or Docker Compose? [swarm/compose]: " docker_type
+done
+
+read -p "Domain name you are deploying to [e.g: 'op-scim.example.com']: " domain_name
+
+while ! [[ -f "$scimsession_file" ]]; do
+    read -p "Path to your scimsession file: " scimsession_file
+    if ! [[ -f "$scimsession_file" ]]
+    then
+        echo "File '$scimsession_file' does not exist at that path, please try again." >&2
+    fi
+done
+
+echo " "
+echo "Using the following parameters to deploy the SCIM Bridge"
+echo "Deployment type:" $docker_type
+echo "scimsession file path:" $scimsession_file
+echo "Domain name:" $domain_name
+
+while ! [[ "$proceed" =~ ^([yY][eE][sS]|[yY])$ || "$proceed" =~ ^([nN][oO][nN]) ]]; do
+    read -p "Does this look correct? [Y/n]: " proceed
+    if [[ "$proceed" =~ ^([nN][oO][nN])$ ]]
+    then
+        echo "Exiting..."
+        exit 1
+    fi
+done
+
+# place the domain name into the deployment file, in a backup
+docker_file_path=$docker_path/$docker_type
+docker_original_file=$docker_file_path/docker-compose.yml
+docker_deploy_file=$docker_file_path/docker-compose.yml.deploy
+sed s/{YOUR-DOMAIN-HERE}/$domain_name/g $docker_original_file > $docker_deploy_file
+
+if [[ "$docker_type" == "compose" ]]
 then
-    run_docker_swarm
-else 
-    echo "Invalid docker manager. Please use either docker-compose or docker-swarm."
-    exit
+    echo " "
+    echo "Deploying using Docker Compose..."
+    echo "(Ctrl+C to cancel)"
+    sleep 3
+
+    # this command populates an .env file which allows the container to have a needed environment variable without needing to store the scimsession file itself
+    SESSION=$(cat $scimsession_file | base64 | tr -d "\n")
+    echo "OP_SESSION=$SESSION" > $docker_path/$docker_type/scim.env
+
+    if ! docker-compose -f $docker_deploy_file up --build -d
+    then
+        echo " "
+        echo "Failed to run docker-compose, please investigate the error"
+        sleep 1
+        exit 1
+    fi
+
+    while ! [[ "$view_logs" =~ ^([yY][eE][sS]|[yY])$ || "$view_logs" =~ ^([nN][oO][nN]) ]]; do
```

Comment:
Can you explain the presence of this conditional to me? Why is the loop needed given the `if` statements below?